lighthouse-core/report/scripts/lighthouse-report.js

```diff
@@ -15,13 +15,81 @@
  * limitations under the License.
  */
 
-/* global window, document */
+/* global ga */
 
 'use strict';
 
-window.addEventListener('DOMContentLoaded', _ => {
-  const printButton = document.querySelector('.js-print');
-  printButton.addEventListener('click', _ => {
-    window.print();
-  });
-});
+const logger = require('./logger');
+
+class LighthouseReport {
+
+  /**
+   * @param {Object=} lhresults Lighthouse JSON results.
+   */
+  constructor(lhresults) {
+    this.onCopy = this.onCopy.bind(this);
+    this.onCopyButtonClick = this.onCopyButtonClick.bind(this);
+
+    this._copyAttempt = false;
+
+    this.json = lhresults || null;
+
+    this._addEventListeners();
+  }
+
+  _addEventListeners() {
+    const printButton = document.querySelector('.js-print');
+    if (printButton) {
+      printButton.addEventListener('click', _ => {
+        window.print();
+      });
+    }
+
+    const copyButton = document.querySelector('.js-copy');
+    if (copyButton) {
+      copyButton.addEventListener('click', this.onCopyButtonClick);
+      document.addEventListener('copy', this.onCopy);
+    }
+  }
+
+  /**
+   * Handler copy events.
+   */
+  onCopy(e) {
+    // Only handle copy button presses (e.g. ignore the user copying page text).
+    if (this._copyAttempt) {
+      // We want to write our own data to the clipboard, not the user's text selection.
+      e.preventDefault();
+      e.clipboardData.setData('text/plain', JSON.stringify(this.json, null, 2));
+      logger.log('Report copied to clipboard');
+    }
+
+    this._copyAttempt = false;
+  }
+
+  /**
+   * Copies the report JSON to the clipboard (if supported by the browser).
+   */
+  onCopyButtonClick() {
+    window.ga && ga('send', 'event', 'report', 'copy');
```

Comment:
to be clear, this will allow us to log copy events in the viewer (where we include ga) but not from other contexts.